src/server.py

```diff
@@ -0,0 +1,106 @@
+# coding=utf-8
+import io
+import mimetypes
+import os
+import socket
+
+
+def server():
+    """Run Server."""
+    server_socket = socket.socket(socket.AF_INET,
+                                  socket.SOCK_STREAM,
+                                  socket.IPPROTO_TCP)
+    server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    address = ('127.0.0.1', 5000)
+    server_socket.bind(address)
+    server_socket.listen(1)
+    conn, addr = server_socket.accept()
+
+    buffer_length = 8
+    message_complete = False
+    return_message = ''
+    while not message_complete:
+        part = conn.recv(buffer_length)
+        if len(part) < buffer_length:
+            message_complete = True
+            server_socket.close()
+        return_message += part.decode('utf-8')
+    print(return_message)
+    try:
+        uri = parse_request(return_message)
+        content = resolve_uri(uri)
+        reply_ok = response_ok()
+        content_type = b'Content Type: ' + content[1] + b'\r\n'
+        content_length = str(len(content[0])).encode('utf-8')
+        content_length = b'Content-Length: ' + content_length + b'\r\n\r\n'
+        reply = reply_ok + content_type + content_length + content[0]
+        print(reply)
+    except NotImplementedError:
+        reply = response_error()
+    except OSError:
+        reply = response_error()
+    conn.sendall(reply)
+    conn.close()
+    server_socket.close()
+    server()
+
+
+def response_ok():
+    """Return a 200 OK response."""
+    ok_200 = b'HTTP/1.1 200 OK\r\n'
+    return ok_200
+
+
+def response_error():
+    """Return a 500 Error response."""
+    no_500 = b'HTTP/1.1 500 Internal Server Error.'
+    return no_500
+
+
+def parse_request(request):
+    """Parse request and return reponse or error."""
+    parsed = request.split()
+    print(parsed)
+    if len(parsed) < 5:
```

Comment:
This is quite an assumption.  What if the Host header isn't the first one? What if there's more than one header in the incoming request?